util/machinecollection/machine_collection.go

```diff
@@ -25,15 +25,15 @@ limitations under the License.
 //   - Updated Has to also check for equality of Machines
 //   - Removed unused methods
 
-package internal
+package machinecollection
```

Comment:
Should we call this `collections` instead?

`machinecollection.FilterableMachineCollection` stutters, might be worthwhile to rename it altogether. Was thinking we could go for `collections.Machines`, we could also add the following:

- Rename `FilterableMachineCollection` to be `Machines`, so it reads `collections.Machines`
- Avoid exposing `Machines` as a plain map, but maybe as a struct?
- `NewFilterableMachineCollection` -> `collections.FromMachines` 
- `NewFilterableMachineCollectionFromMachineList` -> `collections.FromMachineList`


etc

Reply:
- renamed `machinecollection` package to `collections`
- `machinecollection.FilterableMachineCollection` -> `collections.FilterableMachines` (kept filterable to distinguish from other uses of `Machines` in CAPI)
- `NewFilterableMachineCollection` -> `collections.FromMachines`
- `NewFilterableMachineCollectionFromMachineList` -> `collections.FromMachineList`